dask_cloudprovider/providers/azure/azureml.py

```diff
@@ -394,17 +394,19 @@ async def __create_cluster(self):
 
         self.__print_message("Waiting for scheduler node's IP")
         while (
-            run.get_status() != "Canceled"
-            and run.get_status() != "Failed"
+            run.status != "Canceled"
```

Comment:
good change here, but now we need to add a status refresh after the sleep. Consider instead a construction like

```
status = run.get_status()
while status != A and status != B:
    time.sleep(..)
    status = run.get_status()

Reply:
is the refresh needed? run.status show latest state ?

Reply:
to fetch the latest status, you need to use get_status(), which makes a network call

After making a network call, within the scope of a logical unit, you'd want to reuse the information you just got (like run.status or a temporary variable). run.status doesn't get you the latest status

Reply:
updated